src/families/ethereum/nftMetadataResolver.ts

```diff
@@ -0,0 +1,44 @@
+import type { CurrencyBridge } from "../../types";
+
+import { getCryptoCurrencyById } from "../../currencies";
+import { metadataCallBatcher } from "../../nft";
+
+const SUPPORTED_CHAIN_IDS = new Set([
+  1, // Ethereum
+  137, // Polygon
+]);
+
+const nftMetadataResolver: CurrencyBridge["nftMetadataResolver"] = async ({
+  contract,
+  tokenId,
+  currencyId,
+  metadata,
+}) => {
+  // This is for test/mock purposes
+  if (typeof metadata !== "undefined") {
+    return {
+      status: 200,
+      result: {
+        contract,
+        tokenId,
+        ...metadata,
+      },
```

Comment:
Good catch, maybe explicitly set the function's return type to `Promise<NFTMetadataResponse>`, cause I think TS would have seen that.

Reply:
Yeah I don't know why it was catched only now by my VSCode... TS should have told me right away but it wasn't the case when I added the "sync" part of the resolver at the time... 🤔 
And I can't actually be more precise than `CurrencyBridge["nftMetadataResolver"]` here cause it already contains the response part...

Reply:
Yeah idk why but happened to me also that the function's _type_ isn't enough for TS to deduce the _return type_... no idea why, and sometimes it even works.

However I think you could add this to the function's signature, right?
```diff
-  }) => {
+  }): Promise<NFTMetadataResponse> => {
```